refinery/core/views.py

```diff
@@ -693,13 +696,79 @@ def graph(self, request, *args, **kwargs):
         )
 
 
-class NodeViewSet(viewsets.ModelViewSet):
-    """API endpoint that allows Nodes to be viewed"""
-    queryset = Node.objects.all()
-    serializer_class = NodeSerializer
-    lookup_field = 'uuid'
-    http_method_names = ['get']
-    # permission_classes = (IsAuthenticated,)
+class NodeViewSet(APIView):
+    """API endpoint that allows Nodes to be viewed".
+     ---
+    #YAML
+
+    PATCH:
+        parameters_strategy:
+        form: replace
+        query: merge
+
+        parameters:
+            - name: uuid
+              description: User profile uuid used as an identifier
+              type: string
+              paramType: path
+              required: true
+            - name: file_uuid
+              description: uuid for file store item
+              type: string
+              paramType: form
+              required: false
+    ...
+    """
+    http_method_names = ['get', 'patch']
+
+    def get_object(self, uuid):
+        try:
+            return Node.objects.get(uuid=uuid)
+        except Node.DoesNotExist as e:
+            logger.error(e)
+            raise Http404
+        except Node.MultipleObjectsReturned as e:
+            logger.error(e)
+            raise APIException("Multiple objects returned.")
+
+    def get(self, request, uuid):
+        node = self.get_object(uuid)
+        data_set = node.study.get_dataset()
+        public_group = ExtendedGroup.objects.public_group()
+
+        if request.user.has_perm('core.read_dataset', data_set) or \
+                'read_dataset' in get_perms(public_group, data_set):
+            serializer = NodeSerializer(node)
+            return Response(serializer.data)
+
+        return Response(node, status=status.HTTP_401_UNAUTHORIZED)
+
+    def patch(self, request, uuid):
+        node = self.get_object(uuid)
+        old_file_uuid = node.file_uuid
+        data_set_owner = node.study.get_dataset().get_owner()
+
+        if data_set_owner == request.user:
+            serializer = NodeSerializer(node, data=request.data, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+
+                try:
+                    FileStoreItem.objects\
+                        .get(uuid=old_file_uuid).delete_datafile()
```

Comment:
I don't think we would want to delete the datafile on every patch, correct? I would think that would only be necessary if the old file_uuid differed from the new one

Reply:
Yes, correct. I'll fix that.

Reply:
FYI, this api is only supporting what we need which right now is removing file_uuid. It does not support adding or changing.